GroundDBMemoryPool/src/include/rdma.hh

```diff
@@ -41,6 +45,17 @@ struct cm_con_data_t
     uint8_t gid[16]; /* gid */
 } __attribute__((packed));
 
+struct connection{
+    struct ibv_cq *cq;                 /* CQ handle */
+    struct ibv_qp *qp;                 /* QP handle */
+    int sock{-1};                      /* TCP socket file descriptor */
+};
+struct memory_region{
+    std::vector<connection> conns;
+    struct ibv_mr *mr;                 /* MR handle for buf */
+    char *buf;                         /* memory buffer pointer, used for RDMA and send ops */
+    bool buf_exclusive;                /* whether buf is exclusive or not */
+};
```

Comment:
Since one-sided RDMA operations don't involve the CPU of remote memory node, `mutex` might be unuseful in our case. We need to find other ways to implement it such as RDMA atomic operations.

Reply:
Atomicity is a future milestone.